instructionAPI/src/Instruction.C

```diff
@@ -422,7 +421,39 @@ memAccessors.begin()));
       std::copy(m_InsnOp.getImplicitMemWrites().begin(), m_InsnOp.getImplicitMemWrites().end(), std::inserter(memAccessors,
 memAccessors.begin()));
     }
-    
+
+    INSTRUCTION_EXPORT Operand Instruction::getPredicateOperand() const
+    {
+      if (arch_decoded_from != Arch_cuda && m_Operands.empty()) {
+        decodeOperands();
+      }
+
+      for(std::list<Operand>::const_iterator curOperand = m_Operands.begin();
+        curOperand != m_Operands.end(); ++curOperand) {
+        if (curOperand->isTruePredicate() || curOperand->isFalsePredicate()) {
+          //fprintf(stderr, "getPredicateOperand returns %s\n", curOperand->format(Arch_cuda).c_str());
+          return *curOperand;
+        }
+      }
```

Comment:
A range-for makes this a little cleaner:

    for(auto const& op : m_Operands) {
      if (op.isTruePredicate() || op.isFalsePredicate()) {
        return op;
      }
    }